src/profile/v100/app_preferences/gateways/saved_gateways.rs

```diff
@@ -293,4 +294,72 @@ mod tests {
             "#,
         )
     }
+
+    #[test]
+    fn deserialize_from_json_with_different_description_marks_both_gateways_as_wellknown(
```

Comment:
hmm I think `treats` is better than `marks` (marks indicate that a field is changed IMO).